tests/std/tests/GH_000625_vector_bool_optimization/test.cpp

```diff
@@ -253,10 +257,748 @@ void test_huge_vector_bool() {
     assert(count(v.begin(), v.end(), false) == large_bit_diff - 3);
 }
 
+void test_copy_no_offset(const size_t length) {
+    vector<bool> result;
+    // clang-format off
+    switch (length) {
+    case 3:
+        result = { true, false, true };
+        break;
+    case 8:
+        result = { true, false, true, false, true, true, true, false };
+        break;
+    case 22:
+        result = { true, false, true, false, true, true, true, false,
+                   true, false, true, false, true, true, true, false,
+                   true, false, true, false, true, true };
+        break;
+    case 31:
+        result = { true, false, true, false, true, true, true, false,
+                   true, false, true, false, true, true, true, false,
+                   true, false, true, false, true, true, true, false,
+                   true, false, true, false, true, true, true };
+        break;
+    case 32:
+        result = { true, false, true, false, true, true, true, false,
+                   true, false, true, false, true, true, true, false,
+                   true, false, true, false, true, true, true, false,
+                   true, false, true, false, true, true, true, false };
+        break;
+    case 67:
+        result = { true, false, true, false, true, true, true, false,
+                   true, false, true, false, true, true, true, false,
+                   true, false, true, false, true, true, true, false,
+                   true, false, true, false, true, true, true, false,
+                   true, false, true, false, true, true, true, false,
+                   true, false, true, false, true, true, true, false,
+                   true, false, true, false, true, true, true, false,
+                   true, false, true, false, true, true, true, false,
+                   true, false, true };
+        break;
+    default:
+        assert(false);
+    }
+    // clang-format on
+
+    vector<bool> dest(length, false);
+    const auto res_copy = copy(source.begin(), next(source.begin(), length), dest.begin());
+    assert(dest == result);
+    assert(res_copy == dest.end());

```

Comment:
To avoid potential copy-paste errors, and increase the clarity of the code by making it clear that nothing is carried over from each of these sub-tests, it would be good to wrap each of these in braces to create new scopes.